### PR TITLE
Implemented logic for Input Validation and exception handling.

### PR DESCRIPTION
**Pull request by Tanyashrvstv**

### CODE REVIEW

**Comment by Tanyashrvstv** [comment]
Don't merge this PR yet.

File: facad1ng.py
Comment (Himan10) [inline facad1ng.py]: Add the support for http as well. 
Little change in the pattern: `s?`

File: facad1ng.py
Comment (Himan10) [inline facad1ng.py]: You can shorten this Regex pattern as it takes a lot of steps to match the string. 
from the security perspective, suppose if an attacker provides a huge string, then your regex pattern can take a long time to match the text. Hence, results in either time consumption and unexpected behavior by the regex.

File: facad1ng.py
Comment (Himan10) [inline facad1ng.py]: don't perform base64 encoding

File: facad1ng.py
Comment (Himan10) [inline facad1ng.py:112]: Don't use "break" here, we don't have to terminate the program. 
if user enters something wrong, then ask for the input again.

File: facad1ng.py
Comment (Himan10) [inline facad1ng.py]: URL schema doesn't display in the final output. 
image

the URL should start with `http` or `https://`

**Review by Himan10 — CHANGES_REQUESTED** [review]

File: facad1ng.py
Comment (thisisshubhamkumar) [reply facad1ng.py]: Screenshot 2023-11-16 at 8 49 12 PM

URL schema works on the original code.

File: facad1ng.py
Comment (thisisshubhamkumar) [reply facad1ng.py:112]: that we can do.

File: facad1ng.py
Comment (Himan10) [reply facad1ng.py]: yeah, it works in the original code, she probably missed out something. Checking

File: facad1ng.py
Comment (thisisshubhamkumar) [reply facad1ng.py]: This was your idea to hide the vendor name. if not base64 how you gonna hide the shortened URL

File: facad1ng.py
Comment (thisisshubhamkumar) [reply facad1ng.py]: Ok, this we can change but the idea of the tool is to make look phishing URL less suspicious. if the URL starts with "HTTP" it's a kind of red flag.

File: facad1ng.py
Comment (Himan10) [reply facad1ng.py]: the problem with base64 is that the final URL you get won't redirect you to the phishing page. 

For example: this original link `https://ngrok.com`, when converted to base64 looks like this `https://gmail.com-bG9naW4=@aHR0cHM6Ly90aW55dXJsLmNvbS9wcWpzcmU1`. Now the single main issue with this type of url is, when the user pastes it into the browser, the request won't go to the ngrok.com. The base64 encoding of the given url didn't fit into the picture as we imagined it, she checked with the URL encoding as well where special chars gets encoded but it didn't work either. 

We are still working on this, will let you know the possible solutions.

File: facad1ng.py
Comment (Himan10) [reply facad1ng.py]: I do agree with your statement of making it look like less suspicious, but often time the phishing page is hosted on sites where the certificates are self-generated/self-signed or they accept only the http connection, so especially for those scenarios I thought it'd be better to include a support for `http` as well. 

However, if you have don't want this behavior, we can exclude this change.

Comment (thisisshubhamkumar) [reply facad1ng.py]: Good point. we can add `HTTP`

File: facad1ng.py
Comment (thisisshubhamkumar) [reply facad1ng.py]: we can use server-side redirection script to achieve our goal but for this we have to run web server like flask

```
def encode_and_redirect(original_url):
    """
    Encode the original URL and redirect to the server-side endpoint
    """
    s = pyshorteners.Shortener()
    shortened_url = s.tinyurl.short(original_url)
    encoded_url = urlparse(shortened_url).path[1:]
    return redirect(f'/redirect/{encoded_url}')


@app.route('/redirect/<path:encoded_url>')
def redirect_to_original(encoded_url):
    """
    Redirect to the original URL using the decoded URL
    """
    decoded_url = encoded_url.encode('utf-8')
    return redirect(decoded_url)


if __name__ == '__main__':
    print_banners()
    app.run(debug=True)
```

In this example, a Flask web server is used to handle redirection. When you encode a URL using this script, it will create a shortened URL like http://yourserver.com/redirect/encoded_url, and when users visit this shortened URL, they will be redirected to the original URL. This way, you can hide the original URL while still allowing proper redirection.

Not a fan of this method but this can be a solution.

**Comment by Himan10** [comment]
@thisisshubhamkumar you can merge this PR if you want as the primarily objective was to handle exceptions or input cases. 
Will be raising another PR for how to masquerade a URL in such a way that it looks more authentic and legit  without exposing vendor names or similar.

**Pull request merged**
